tensilelite/rocisa/rocisa/include/helper.hpp

```diff
@@ -47,50 +49,87 @@ inline std::string getGfxNameTuple(const IsaVersion& isaVersion)
            + int_to_hex[isaVersion[2]];
 }
 
-inline std::pair<int, std::string>
-    run(const std::vector<char*>& cmd, const std::string& input, bool debug = false)
+#define CHECK_COMGR(call)                                                                  \
+    do                                                                                     \
+    {                                                                                      \
+        amd_comgr_status_t status = call;                                                  \
+        if(status != AMD_COMGR_STATUS_SUCCESS)                                             \
+        {                                                                                  \
+            const char* status_string = nullptr;                                           \
+            amd_comgr_status_string(status, &status_string);                               \
+            throw std::runtime_error(std::string("COMGR call failed: ") + #call            \
+                                     + " (COMGR status: " + std::to_string(status) + ", "  \
+                                     + (status_string ? status_string : "Unknown") + ")"); \
+        }                                                                                  \
+    } while(0)
+
+inline std::pair<int, std::string> run(const std::string& input,
```

Comment:
Tip: if ever marking big functions like this inline just so you can define them in a header vs impl file, that is a sign they are misplaced.